confgenerator/logging_receivers.go

```diff
@@ -125,7 +125,7 @@ func (r LoggingReceiverSyslog) Components(tag string) []fluentbit.Component {
 			"Mode":   r.TransportProtocol,
 			"Listen": r.ListenHost,
 			"Port":   fmt.Sprintf("%d", r.ListenPort),
-			"Parser": tag,
+			"Parser": "syslog-rfc5424",
```

Comment:
If memcached outputs to syslog by default that probably means it writes to `/dev/log`, which means it will end up in `journald` and that's where you need to read the logs from. Using a syslog receiver in Ops Agent would mean that the user would have to reconfigure memcached to send logs to a different port that OA is listening on.

Reply:
Ok, so we tested and it doesn't end up in journald just syslog, I've actually reverted this change because we originally thought that the parser was going to properly parse the different parts of the logs and double checked it and there doesn't seem to be any benefit.

Reply:
So with an out-of-the-box Ops Agent and an 
out-of-the-box Memcached today, logs are ingested but not parsed as structured logs, right?

Can we add a "format" parameter to the out-of-the-box ops agent syslog receiver and document sth like:

Memcached logs are collected automatically as part of syslog. To enable structured logging for all syslog, use:

```
logging:
  receivers:
    syslog:
      format:
        syslog-rfc5424
```

Reply:
`syslog-rfc5424` is a network protocol; it applies to `type: syslog` receivers, which the default `syslog` receiver is not.

AFAICT memcached just sends its logs to stderr, so there's no `syslog`-type receiver needed. If it's running as a normal systemd service, you would want to use a journald receiver to extract its logs.

@Mrod1598 Maybe we can schedule a 1:1 where you walk me through what you tested?

Reply:
> So with an out-of-the-box Ops Agent and an out-of-the-box Memcached today, logs are ingested but not parsed as structured logs, right?
> 
> Can we add a "format" parameter to the out-of-the-box ops agent syslog receiver and document sth like:
> 
> Memcached logs are collected automatically as part of syslog. To enable structured logging for all syslog, use:
> 
> ```
> logging:
>   receivers:
>     syslog:
>       format:
>         syslog-rfc5424
> ```
> 
> This way it's backward compatible with existing syslog functionality.
> 
> To enable Memcached specific parsing, we can do an app specific processor to avoid duplicate logs while providing the functionality and maintaining backward compatibility. e.g.
> 
> ```
> logging:
>   receivers:
>     syslog:
>       format:
>         syslog-rfc5424
>   processors:
>     parse_memcached:
>         type: parse_memcached
>   pipelines:
>     default_pipeline:
>       processors:
>         parse_memcached
> ```
> 
> This is not a blocker for the metrics side of things though IMO.

These are the logs I'm getting with the new parser vs. no change

## No Change Logs
```
{
  "insertId": "rh0siyfvqwreh",
  "jsonPayload": {
    "message": "Dec 13 20:14:35 test-memcached-no-change-logs systemd-memcached-wrapper[13698]: <27 set foo 0 0 8"
  },
  "resource": {
    "type": "gce_instance",
    "labels": {
      "instance_id": "2603025410381497601",
      "project_id": "otel-agent-dev",
      "zone": "us-central1-a"
    }
  },
  "timestamp": "2021-12-13T20:14:35.980433749Z",
  "logName": "projects/otel-agent-dev/logs/syslog",
  "receiveTimestamp": "2021-12-13T20:14:36.102239290Z"
}
```
## Updated Parser Logs
```
{
  "insertId": "1vzkvaifr7z3ak",
  "jsonPayload": {
    "message": "Dec 13 20:20:58 test-memcached--new-parser-logs systemd-memcached-wrapper[13467]: <27 set foo 0 0 8"
  },
  "resource": {
    "type": "gce_instance",
    "labels": {
      "instance_id": "5992099982425637625",
      "project_id": "otel-agent-dev",
      "zone": "us-central1-a"
    }
  },
  "timestamp": "2021-12-13T20:21:40.089486747Z",
  "logName": "projects/otel-agent-dev/logs/syslog",
  "receiveTimestamp": "2021-12-13T20:21:41.041395584Z"
}
```
I think we were incorrect the first time we tested this. I don't see a difference and it just pulls it by default with the ops-agent

Reply:
Per sync offline, this PR will focus on metrics only. For logging, we'll rely on the out-of-the-box `syslog` receiver to collect Memcached logs and add processors as seen fit in a separate effort.